src/Components/JoinScreen/WaitingRoom.tsx

```diff
@@ -0,0 +1,71 @@
+import { ReactElement, useEffect, useState } from "react";
+
+import Button from '@mui/material/Button';
+import Card from '@mui/material/Card';
+import CardActions from '@mui/material/CardActions';
+import CardContent from '@mui/material/CardContent';
+import { Grid } from "@mui/material";
+import Typography from '@mui/material/Typography';
+import socket from "../Websocket/socket";
+
+type Player = {
+  username: string,
+  status: boolean
```

Comment:
You can set the status type to type 'ready' | 'not ready' or something like that. That way typescript will only allow the strings 'ready' or 'not ready'.